synapse/federation/federation_client.py

```diff
@@ -250,6 +250,8 @@ async def backfill(
 
```

Comment:
Neither here nor there but it bothers me that we use `__bool__` on `extremities` above, which is a generic `Iterable`. It happens in the other `backfill` methods too.